cmake/traccc-functions.cmake

```diff
@@ -146,3 +146,35 @@ function( traccc_add_flag name value )
    set( ${name} "${${name}} ${value}" PARENT_SCOPE )
 
 endfunction( traccc_add_flag )
+
+# from https://stackoverflow.com/a/52136398
+# necessary until cmake 3.25 https://cmake.org/cmake/help/latest/prop_tgt/SYSTEM.html#prop_tgt:SYSTEM
+
+function(traccc_target_link_libraries_system target)
+  set(options PRIVATE PUBLIC INTERFACE)
+  cmake_parse_arguments(TLLS "${options}" "" "" ${ARGN})
+  foreach(op ${options})
+    if(TLLS_${op})
+      set(scope ${op})
+    endif()
+  endforeach(op)
+  set(libs ${TLLS_UNPARSED_ARGUMENTS})
+
+  foreach(lib ${libs})
+    get_target_property(lib_include_dirs ${lib} INTERFACE_INCLUDE_DIRECTORIES)
+    if(lib_include_dirs)
+      if(scope)
+        target_include_directories(${target} SYSTEM ${scope} ${lib_include_dirs})
+      else()
+        target_include_directories(${target} SYSTEM PRIVATE ${lib_include_dirs})
+      endif()
+    else()
+      message("Warning: ${lib} doesn't set INTERFACE_INCLUDE_DIRECTORIES. No include_directories set.")
+    endif()
+    if(scope)
+      target_link_libraries(${target} ${scope} ${lib})
+    else()
+      target_link_libraries(${target} ${lib})
+    endif()
+  endforeach()
+endfunction(acts_target_link_libraries_system)
```

Comment:
This is an interesting way of doing it, but I'd do it in a different way... :thinking:

For a number of externals we already had to do a bit of massaging to have them appear as system headers in our build. Even though we build them as part of this project. See:
  - https://github.com/acts-project/traccc/blob/main/extern/eigen3/CMakeLists.txt#L48-L53
  - https://github.com/acts-project/algebra-plugins/blob/main/extern/vc/CMakeLists.txt#L48-L50
  - etc.

I'm not actually sure why we don't use the more compact formalism from algebra-plugins for Eigen as well. :confused: Probably that's my bad...

Point being, with that setup the "main CMake code" should not have to do anything fancy. When we take an external "from the system" with `find_package(...)`, its headers will be treated as system headers with older CMake versions as well. And when we build the external ourselves, this massaging still allows the CMake code of the main libraries to be none the wiser. :thinking:

Reply:
I don't think it's a good idea at all to modify targets from `FetchContent` at all, this blurs the line between what the child project does and what the parent project does. 

Adding it as an explicit argument when linking against the library in question is cleaner and conveys + documents what's going on.

Reply:
Anyway, not a hill I'm willing to die on, so I'll change it.